src/corehost/corehost.cpp

```diff
@@ -293,9 +293,20 @@ int main(const int argc, const pal::char_t* argv[])
 #if defined(_WIN32) && defined(FEATURE_APPHOST)
     if (get_windows_graphical_user_interface_bit())
     {
-        // If this is a GUI application, buffer errors to display them later. Without this any errors are effectively lost
-        // unless the caller explicitly redirects stderr. This leads to bad experience of running the GUI app and nothing happening.
-        trace::set_error_writer(buffering_trace_writer);
+        pal::string_t env_value;
+        bool gui_errors_disabled = false;
+
+        if (pal::getenv(_X("DOTNET_DISABLE_GUI_ERRORS"), &env_value))
+        {
+            gui_errors_disabled = pal::xtoi(env_value.c_str()) == 1;
+        }
+
+        if (!gui_errors_disabled)
+        {
+            // If this is a GUI application, buffer errors to display them later. Without this any errors are effectively lost
+            // unless the caller explicitly redirects stderr. This leads to bad experience of running the GUI app and nothing happening.
```

Comment:
> Without this any errors are effectively lost unless the caller explicitly redirects stderr

Maybe I'm misinterpreting the comment, but does this mean that if `DOTNET_DISABLE_GUI_ERRORS` is set, the only way to get an error log is through setting `COREHOST_TRACE_FILE`?

Reply:
Not only - it is perfectly valid to run a GUI app with redirected `stdout` and `stderr`. In fact you can see it in the tests, it runs the GUI app with this setting on and captures the error via normal `stderr` redirection.

Since this targets mostly test infrastructures, those are pretty likely to redirect out/err anyway.